src/SignalR/server/Core/src/Internal/Utf8HashLookup.cs

```diff
@@ -81,32 +109,52 @@ private bool TryGetValue(ReadOnlySpan<char> key, [MaybeNullWhen(false), AllowNul
         return false;
     }
 
-    private static int GetKeyHashCode(ReadOnlySpan<char> key)
+    private static int GetHashCode(ReadOnlySpan<char> value) =>
+        HashCodeMask & string.GetHashCode(value, StringComparison.OrdinalIgnoreCase);
+
+    private static int GetCaseSensitiveHashCode(ReadOnlySpan<byte> encodedValue)
     {
-        return HashCodeMask & string.GetHashCode(key, StringComparison.OrdinalIgnoreCase);
+        var hashCode = new HashCode();
+        hashCode.AddBytes(encodedValue);
```

Comment:
I was looking at this originally and wondering if we should do marvin hash (what string does) here. (https://source.dot.net/#System.Private.CoreLib/Marvin.cs,17)

Reply:
I don't think it's necessary because the cached Hub method names come from the application not the client, so it shouldn't be a DoS vector.

Reply:
Worth consulting @GrabYourPitchforks and @blowdart on.

Reply:
Using the _HashCode_ class will already give you randomization, so whatever. :)

> I don't think it's necessary because the cached Hub method names come from the application not the client, so it shouldn't be a DoS vector.

This statement makes sense to me.